modules/consul-cluster/main.tf

```diff
@@ -51,6 +51,34 @@ resource "aws_autoscaling_group" "autoscaling_group" {
       var.tags,
     ]
   )
+  
+  dynamic "initial_lifecycle_hook" {
+    for_each = var.lifecycle_hook_launching
+    
+    content {
+      name                    = initial_lifecycle_hook.key
+      default_result          = lookup(initial_lifecycle_hook.value,"default_result",null)
+      heartbeat_timeout       = lookup(initial_lifecycle_hook.value,"heartbeat_timeout",null)
+      lifecycle_transition    = "autoscaling:EC2_INSTANCE_LAUNCHING"
```

Comment:
It was to make the object passed into the module look as close to the lifecycle_hook terraform resource as possible so the user of the module didn't need to explicitly put "heartbeat_timeout=null, default_result=null" etc.

For terraform *resources*, I think optional parameters can be missing - but referencing a missing property from the object will cause terraform to error.

While a little less convenient, it would be more declarative to reference them explicitly.  Maybe few enough people would be exposed to putting nulls in for it not to matter?

Reply:
I think a single input variable and `dynamic` block will be easier to understand/maintain. Perhaps you could use a [validation rule](https://www.terraform.io/docs/language/values/variables.html#custom-validation-rules) to check the `lifecycle_transition` value?

Reply:
Thanks - that's done now: a single block and validation rule on those properties.